From: https://bugs.debian.org/933151
Author: Aurelien Jarno <aurelien@aurel32.net>

Ubuntu has patched their version of cmake to link with -latomic on
riscv64. While patching cmake is a really good idea, the fix is wrong,
the correct think to do is to link with -pthread instead of -lpthread,
and do that for all architectures.

This is the strategy followed for mariadb-10.5 in the attached patch. I
have tested it and it builds fine on Debian.

--- a/configure.cmake
+++ b/configure.cmake
@@ -134,6 +134,7 @@ IF(UNIX)
   IF(NOT LIBRT)
     MY_SEARCH_LIBS(clock_gettime rt LIBRT)
   ENDIF()
+  set(THREADS_PREFER_PTHREAD_FLAG ON)
   FIND_PACKAGE(Threads)
 
   SET(CMAKE_REQUIRED_LIBRARIES 
